odootil/models/odootil.py

```diff
@@ -0,0 +1,617 @@
+import os
+import six
+from num2words import num2words
+import operator
+import validators
+
+from odoo import models, api, fields, _
+from odoo.tools.convert import convert_file
+from odoo.exceptions import ValidationError, UserError
+
+from odoo.addons.base.models.ir_sequence import IrSequence
+
+# Common constants.
+PRIORITY = [(1, 'Low'), (2, 'Normal'), (3, 'High')]
+DATE_FMT = '%Y-%m-%d'
+DATETIME_FMT = '%Y-%m-%d %H:%M:%S'
+
+# Cache for singleton records.
+singleton_records = {}
+
+
+class Interpolation(object):
+    """Interpolate string using datetime variables.
+
+    Attaching already existing implementation from ir.sequence model.
+    """
+
+    def __init__(self, prefix='', suffix='', _name='', _context=None):
+        """Initialize Interpolation class for string rendering."""
+        self.prefix = prefix
+        self.suffix = suffix
+        self._name = _name
+        self._context = _context or {}
+
+    # Need to implement this method, because _get_prefix_suffix expects
+    # it, so we mimic that.
+    def get(self, name):
+        """Implement odoo model's 'get' method interface."""
+        # name argument is ignored here actually, we only pass it,
+        # because of attached method requirement.
+        return self._name
+
+    get_prefix_suffix = IrSequence._get_prefix_suffix
+
+
+class Odootil(models.AbstractModel):
+    """Model to add some common functionality patterns for modules.
+
+    Provided methods can be used as helpers for better re-usability.
+    """
+
+    _name = 'odootil'
+    _description = 'Odoo Utilities'
+
+    # Check helpers.
+
+    # TODO: update translations after module is finished being migrated.
+    # TODO: rename to something like find_dupes/find_dupes_count (where
+    # count method could be used for just count search.)
+    # TODO: refactor to again use full domain instead of one value and
+    # field name.
+    # TODO: refactor this once related modules are being migrated.
+    def check_field_unique(
+        self,
+        Model,
```

Comment:
I guess I have older package, cause dont get any of that. It will be removed later, because intention is to use on base, so won't need to specify Model.